ghost/email-service/lib/email-service.js

```diff
@@ -127,14 +135,87 @@ class EmailService {
         return email;
     }
 
+    /**
+     * @private
+     * @param {string} [email] (optional) Search for a member with this email address and use it as the example. If not found, defaults to the default but still uses the provided email address.
+     * @return {Promise<import('./email-renderer').MemberLike>}
+     */
+    async getExampleMember(email) {
+        /**
+         * @type {import('./email-renderer').MemberLike}
+         */
+        const exampleMember = {
+            id: 'example-id',
+            uuid: 'example-uuid',
+            email: 'jamie@example.com',
+            name: 'Jamie Larson'
+        };
+
+        // fetch any matching members so that replacements use expected values
+        if (email) {
+            const member = await this.#membersRepository.get({email});
+            if (member) {
+                exampleMember.id = member.id;
+                exampleMember.uuid = member.get('uuid');
+                exampleMember.email = member.get('email');
+                exampleMember.name = member.get('name');
+            } else {
+                exampleMember.email = email;
+            }
+        }
+
+        return exampleMember;
+    }
+
+    /**
+     * 
+     * @param {*} post 
+     * @param {*} newsletter 
+     * @param {import('./email-renderer').Segment} segment 
+     * @returns {Promise<{subject: string, html: string, plaintext: string}>} Email preview
+     */
     async previewEmail(post, newsletter, segment) {
-        // eslint-disable-next-line no-restricted-syntax
-        throw new Error('Previewing an email has not been implemented yet. Turn off the email stability flag is you need this functionality.');
+        const exampleMember = await this.getExampleMember();
+
+        const subject = this.#emailRenderer.getSubject(post);
+        let {html, plaintext, replacements} = await this.#emailRenderer.renderBody(post, newsletter, segment, {clickTrackingEnabled: false});
+
+        // Do manual replacements with an example member
+        for (const replacement of replacements) {
+            html = html.replace(replacement.token, replacement.getValue(exampleMember));
+            plaintext = plaintext.replace(replacement.token, replacement.getValue(exampleMember));
+        }
+
+        return {
+            subject,
+            html,
+            plaintext
+        };
     }
 
+    /**
+     * 
+     * @param {*} post 
+     * @param {*} newsletter 
+     * @param {import('./email-renderer').Segment} segment 
+     * @param {string[]} emails
+     */
     async sendTestEmail(post, newsletter, segment, emails) {
-        // eslint-disable-next-line no-restricted-syntax
-        throw new Error('Sending a test email has not been implemented yet. Turn off the email stability flag is you need this functionality.');
+        const members = [];
+        for (const email of emails) {
+            members.push(await this.getExampleMember(email));
```

Comment:
🤔 Wondering if its a good idea to use dummy values for `name` when someone enters an email that doesn't match a valid member on site, as the current `getExampleMember` method will always attach a member name to the member data which is not ideal while testing fallbacks on the email.

Reply:
Good point, I'll remove the default name :)

Reply:
Or they could enter a member email address that doesn't have a name to test fallbacks 🤔